pkg/config/config.go

```diff
@@ -1,26 +1,98 @@
 package config
 
+import (
+	"fmt"
+
+	"github.com/caarlos0/env"
+	"github.com/golang/glog"
+)
+
 type DaemonConfig struct {
-	Plugin         string `json:"plugin"`          // Subnet manager plugin name
-	PeriodicUpdate int    `json:"periodic_update"` // Interval between every check for the added and deleted pods
-	GuidPool       GuidPoolConfig
+	PeriodicUpdate int    `env:"PERIODIC_UPDATE"` // Interval between every check for the added and deleted pods
+	RangeStart     string `env:"RANGE_START"`     // First guid of the pool
```

Comment:
Why you move it to be flat?

Reply:
The `GuidPool` is not used any other places, even when pass the startRange and endRange when create the guidPool they are passed as arguments not as struct.

Reply:
So the guid pool should get the config. In the future we may add config and we dont need to to change the signature of the method

Reply:
I will change the newGuidPool to recevie the GuidPoolConfig struct instead of flat args

Reply:
@moshe010 changed